app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java

```diff
@@ -56,84 +56,78 @@
     private int toDay;
     private int toMonth;
     private int toYear;
-    private DatabaseHandler dB;
-    private MainActivity activity;
-
-    public ExportPresenter(MainActivity exportActivity, DatabaseHandler dbHandler) {
-        this.activity = exportActivity;
-        dB = dbHandler;
-    }
 
-    public void onExportClicked(final Boolean all) {
-        final List<GlucoseReading> readings;
+    private Activity mActivity;
+    private ExportView mExportView;
+    private DatabaseHandler dB;
 
-        if (all) {
-            readings = dB.getGlucoseReadings();
+    public ExportPresenter(Activity activity) {
+        mActivity = activity;
+        if (activity instanceof ExportView) {
+            mExportView = (ExportView) activity;
         } else {
-            Calendar fromDate = Calendar.getInstance();
-            fromDate.set(Calendar.YEAR, fromYear);
-            fromDate.set(Calendar.MONTH, fromMonth);
-            fromDate.set(Calendar.DAY_OF_MONTH, fromDay);
-
-            Calendar toDate = Calendar.getInstance();
-            toDate.set(Calendar.YEAR, toYear);
-            toDate.set(Calendar.MONTH, toMonth);
-            toDate.set(Calendar.DAY_OF_MONTH, toDay);
-            readings = dB.getGlucoseReadings(fromDate.getTime(), toDate.getTime());
+            throw new RuntimeException("ExportPresenter Activity must implement ExportView interface");
         }
+        dB = GlucosioApplication.getInstance().getDBHandler();
```

Comment:
This is actually breaking testability. If you have parameter then you can easy mock it in unit tests. You have to make static setters or some hooks with the singleton pattern.